src/cpu_feature/arm_linux/auxv.rs

```diff
@@ -0,0 +1,313 @@
+
+                    read_bytes += n;
+                }
+                Err(_) => return Err(AuxValError::IoError)
+            }
+        }
+
+        let mut reader = &buf[..];
+        let found_aux_type = T::read::<B>(&mut reader)
+            .map_err(|_| AuxValError::InvalidFormat)?;
+        let aux_val = T::read::<B>(&mut reader)
+            .map_err(|_| AuxValError::InvalidFormat)?;
+
+        if aux_types.contains(&found_aux_type) {
+            let _ = result.insert(found_aux_type, aux_val);
+        }
+
+        // AT_NULL (0) signals the end of auxv
+        if found_aux_type == T::from(0) {
+            return Ok(result);
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod tests {
+    extern crate byteorder;
+
+    use std::path::Path;
+    use super::{AuxValError, AuxvTypes, search_procfs_auxv};
+    #[cfg(target_os="linux")]
+    use super::{AuxvUnsignedLongNative, GetauxvalProvider,
+        NativeGetauxvalProvider};
+
+    use self::byteorder::LittleEndian;
+
+    // uid of program that read /proc/self/auxv
+    const AT_UID_64: u64 = 11;
+    const AT_UID_32: u32 = 11;
+
+    // x86 hwcap bits from [linux]/arch/x86/include/asm/cpufeature.h
+    const X86_FPU: u32 = 0 * 32 + 0;
+    const X86_ACPI: u32 = 0 * 32 + 22;
+
+    #[test]
+    #[cfg(target_os="linux")]
+    fn test_getauxv_hwcap_linux_finds_hwcap() {
+        let native_getauxval = NativeGetauxvalProvider{};
+        let result = native_getauxval.getauxval(AuxvTypes::new().AT_HWCAP);
+        assert!(result.is_some());
+        // there should be SOMETHING in the value
+        assert!(result.unwrap() > 0);
+    }
+
+    #[test]
+    #[cfg(target_os="linux")]
+    fn test_getauxv_hwcap_linux_doesnt_find_bogus_type() {
+        let native_getauxval = NativeGetauxvalProvider{};
+
+        assert!(native_getauxval.getauxval(
+            AuxvUnsignedLongNative::from(555555555_u32)).is_none());
+    }
+
+    #[test]
+    fn test_parse_auxv_virtualbox_linux() {
+        let path = Path::new("src/cpu_feature/arm_linux/test-data/macos-virtualbox-linux-x64-4850HQ.auxv");
+        let vals = search_procfs_auxv::<u64, LittleEndian>(path,
+                                                           &[u64_auxv_types().AT_HWCAP,
+                                                               u64_auxv_types().AT_HWCAP2,
+                                                               AT_UID_64])
+            .unwrap();
+        let hwcap = vals.get(&u64_auxv_types().AT_HWCAP).unwrap();
+        assert_eq!(&395049983_u64, hwcap);
+
+        assert_eq!(1, 1 << X86_FPU & hwcap);
+        // virtualized, no acpi via msr I guess
+        assert_eq!(0, 1 << X86_ACPI & hwcap);
+
+        assert!(!vals.contains_key(&u64_auxv_types().AT_HWCAP2));
+
+        assert_eq!(&1000_u64, vals.get(&AT_UID_64).unwrap());
+    }
+
+    #[test]
+    fn test_parse_auxv_virtualbox_linux_32bit() {
+        let path = Path::new("src/cpu_feature/arm_linux/test-data/macos-virtualbox-linux-x86-4850HQ.auxv");
```

Comment:
I was unable to find 32-bit arm auxv files, so if you've got any, that would be great!